crates/accelerate/src/basis/basis_translator/mod.rs

```diff
@@ -10,14 +10,804 @@
+    }
+
+    // Recurse for QuantumCircuit
+    fn recurse_circuit(
+        py: Python,
+        circuit: Bound<PyAny>,
+        basis: &mut HashSet<GateIdentifier>,
+        min_qubits: usize,
+    ) -> PyResult<()> {
+        let circuit_data: PyRef<CircuitData> = circuit
+            .getattr(intern!(py, "_data"))?
+            .downcast_into()?
+            .borrow();
+        for (index, inst) in circuit_data.iter().enumerate() {
+            let instruction_object = circuit.get_item(index)?;
+            let has_calibration =
+                circuit.call_method1(intern!(py, "has_calibration_for"), (&instruction_object,))?;
+            if !has_calibration.is_truthy()?
+                && circuit_data.get_qargs(inst.qubits).len() >= min_qubits
+            {
+                basis.insert((inst.op.name().to_string(), inst.op.num_qubits()));
+            }
+            if inst.op.control_flow() {
+                let operation_ob = instruction_object.getattr(intern!(py, "operation"))?;
+                let blocks = operation_ob.getattr("blocks")?;
+                for block in blocks.iter()? {
+                    recurse_circuit(py, block?, basis, min_qubits)?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    recurse_dag(py, circuit, &mut basis, min_qubits)?;
+    Ok(basis)
+}
+
+/// Method that extracts a mapping of all the qargs in the local_source basis
+/// obtained from the [Target], to all non-calibrated gate instances identifiers from a DAGCircuit.
+/// When dealing with `ControlFlowOp` instances the function will perform a recursion call
+/// to a variant design to handle instances of `QuantumCircuit`.
+fn extract_basis_target(
+    py: Python,
+    dag: &DAGCircuit,
+    source_basis: &mut HashSet<GateIdentifier>,
+    qargs_local_source_basis: &mut HashMap<Option<Qargs>, HashSet<GateIdentifier>>,
+    min_qubits: usize,
+    qargs_with_non_global_operation: &HashMap<Option<Qargs>, HashSet<String>>,
+) -> PyResult<()> {
+    for node in dag.op_nodes(true) {
+        let node_obj: &PackedInstruction = dag.dag()[node].unwrap_operation();
+        let qargs: &[Qubit] = dag.get_qargs(node_obj.qubits);
+        if dag.has_calibration_for_index(py, node)? || qargs.len() < min_qubits {
+            continue;
+        }
+        // Treat the instruction as on an incomplete basis if the qargs are in the
+        // qargs_with_non_global_operation dictionary or if any of the qubits in qargs
+        // are a superset for a non-local operation. For example, if the qargs
+        // are (0, 1) and that's a global (ie no non-local operations on (0, 1)
+        // operation but there is a non-local operation on (1,) we need to
+        // do an extra non-local search for this op to ensure we include any
+        // single qubit operation for (1,) as valid. This pattern also holds
+        // true for > 2q ops too (so for 4q operations we need to check for 3q, 2q,
+        // and 1q operations in the same manner)
```

Comment:
Is there a chance that the original comment (I know this comes from python) mixes up `non-global` and `non-local`?  I would expect the following, but I might be misunderstanding it:

```suggestion
        // Treat the instruction as on an incomplete basis if the qargs are in the
        // qargs_with_non_global_operation dictionary or if any of the qubits in qargs
        // are a superset for a non-global operation. For example, if the qargs
        // are (0, 1) and that's a global operation (i.e. no non-global operations
        //  on (0, 1)), but there is a non-global operation on (1,), we need to
        // do an extra non-global search for this op to ensure we include any
        // single qubit operation for (1,) as valid. This pattern also holds
        // true for > 2q ops too (so for 4q operations we need to check for 3q, 2q,
        // and 1q operations in the same manner)
```

Reply:
This would also apply to L367

Reply:
There is a chance. I particularly don't have a lot of knowledge of what this means but I believe the intention was something like this:
- Local gates might be non-global gates (i.e. Gates that can't be used with all the qargs couplings in the `Target` as long as they match the number of qubits required by the gate).
- Global gates might be the non-local ones (i.e. Gates that can be used with any qubit coupling in the `Target` as long as they match the number of qubits required by the gate).

I'm still iffy about this being the intended but that's what I understood from it.

Reply:
Ok, I re-read it and I think the original comment is correct. I probably got confused with the double negations.